frame/staking/src/lib.rs

```diff
@@ -1484,18 +1531,31 @@ decl_module! {
 		///
 		/// # <weight>
 		/// - The transaction's complexity is proportional to the size of `targets`,
-		/// which is capped at CompactAssignments::LIMIT.
+		/// which is capped at CompactAssignments::LIMIT (MAX_NOMINATIONS).
 		/// - Both the reads and writes follow a similar pattern.
+		/// ---------
+		/// Base Weight: 22.34 + .36 * n µs
+		/// DB Weight:
+		/// - Reads: Era Election Status, Ledger, Current Era
+		/// - Writes: Validators, Nominators
 		/// # </weight>
-		#[weight = 750_000_000]
+		#[weight = FunctionOf(
+			|(targets,): (&Vec<<T::Lookup as StaticLookup>::Source>,)| {
+				T::DbWeight::get().reads_writes(3, 2)
+					.saturating_add(22 * WEIGHT_PER_MICROS)
+					.saturating_add((360 * WEIGHT_PER_NANOS).saturating_mul(targets.len() as Weight))
+			},
+			DispatchClass::Normal,
+			Pays::Yes
+		)]
 		pub fn nominate(origin, targets: Vec<<T::Lookup as StaticLookup>::Source>) {
 			ensure!(Self::era_election_status().is_closed(), Error::<T>::CallNotAllowed);
 			let controller = ensure_signed(origin)?;
 			let ledger = Self::ledger(&controller).ok_or(Error::<T>::NotController)?;
 			let stash = &ledger.stash;
 			ensure!(!targets.is_empty(), Error::<T>::EmptyTargets);
 			let targets = targets.into_iter()
-				.take(<CompactAssignments as VotingLimit>::LIMIT)
+				.take(MAX_NOMINATIONS)
```

Comment:
where is this defined?

Reply:
from here https://github.com/paritytech/substrate/blob/fe67a0c41884bda918b55b658a041588a5014a3e/frame/staking/src/lib.rs#L332

Reply:
I changed it because it is easier to read and understand in the code